docker/Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM python:3.10-slim-bullseye
+
+# The MAPPROXY_VERSION argument could be used like this to overwrite the default:
+# docker build --build-arg MAPPROXY_VERSION=1.15.1 -t mapproxy:1.15.1-python3.10 .
+ARG MAPPROXY_VERSION=1.15.1
+
+# If you want the image to contain an nginx reverse proxy, set this variable to 1.
+# Example:
+# docker build --build-arg NGINX=1 -t mapproxy:1.15.1-python3.10 .
+ARG NGINX=0
+ENV NGINX=${NGINX}
+
+# This flag decides if the image should start mapproxys dev-server on startup
+# Only works when not using nginx from above (NGINX=0).
+# If set to 1, it will launch the dev server on port 8080, which you need to expose to the host.
+# Example:
+# docker build --build-arg MAPPROXY_VERSION=1.15.1 --build-arg DEVSERVER=1 -t mapproxy:1.15.1-python3.10 .
+ARG DEVSERVER=0
+ENV DEVSERVER=${DEVSERVER}
+
+RUN apt update && apt -y install --no-install-recommends \
+  python3-pil \
+  python3-yaml \
+  python3-pyproj \
+  libgeos-dev \
+  python3-lxml \
+  libgdal-dev \
+  python3-shapely \
+  libxml2-dev libxslt-dev
+
+RUN if [ "$NGINX" = '1' ]; then \
```

Comment:
Does it make sense to rather use multi stage builds and the `target` flag, for better isolation of the different builds?

The first stage could simply install mapproxy, the second stage install nginx, the third runs the dev setup, ...

Reply:
yes good point, will look into it. i think it will also reduce final image size